spotbugs/src/main/java/edu/umd/cs/findbugs/detect/ThrowingExceptions.java

```diff
@@ -161,54 +189,33 @@ private boolean sameSignature(Method child, Method parent) {
         }
 
         String sig = child.getSignature();
-        int genIndex = 1;
-        char genSigChar = genSig.charAt(genIndex);
-        int index = 1;
-        char sigChar = sig.charAt(index);
-        while (genSigChar != ')' && sigChar != ')') {
-            switch (genSigChar) {
-            default:
-                if (sigChar != genSigChar) {
-                    return false;
-                }
-                ++index;
-                ++genIndex;
-                break;
-            case 'L':
-                int genSigObjEnd = genSig.indexOf(';', genIndex);
-                int sigObjEnd = sig.indexOf(';', index);
-                if (!genSig.substring(genIndex, genSigObjEnd - genIndex)
-                        .equals(sig.substring(index, sigObjEnd - index))) {
+
+        SignatureParser genSP = new SignatureParser(genSig);
+        SignatureParser sp = new SignatureParser(sig);
+
+        if (genSP.getNumParameters() != sp.getNumParameters()) {
+            return false;
+        }
+
+        String[] gArgs = genSP.getArguments();
+        String[] args = sp.getArguments();
+
+        for (int i = 0; i < sp.getNumParameters(); ++i) {
+            if (gArgs[i].charAt(i) == 'T') {
```

Comment:
Is it impossible to have another char instead of `T`? If possible, our code needs to be more generic to support.

Reply:
`T` means that the type of the parameter or the return value is a generic parameter of the method or the class. For such parameters we do not expect a full match but for the rest we do.